core/src/main/kotlin/su/plo/voice/discs/event/JukeboxEventListener.kt

```diff
@@ -249,6 +252,7 @@ class JukeboxEventListener : Listener, PluginKoinComponent {
 
                 plugin.suspendSync(block.location) {
                     val jukebox = block.asJukebox() ?: return@suspendSync
+                    job.takeIf { !it.isCancelled } ?: return@suspendSync
```

Comment:
This doesn't make sense, the job was cancelled few lines above and `job#isCancelled` will be always `false`.

To fix the issue it's better to check if the current job is equals to the job in the map, this way only correct job will be removed from the map.
```suggestion
                    val currentJob = jobByBlock[block] ?: return@suspendSync
                    if (currentJob != this@launch) return@suspendSync
```

Reply:
`job#isCaneclled` is true in case if the disc got ejected. I used it to confirm that the audio didn't stop playing by ejecting the disc.
What you wrote seems to look better though.

Reply:
Ohh, I'm feel stupid. If disc is ejected the job is cancelled. If track is just ended, then it's not cancelled. Make sense now.
Well, both solutions are viable I guess.